meta1v2/meta1_backend_services.c

```diff
@@ -629,45 +623,61 @@ __get_container_service2(struct sqlx_sqlite3_s *sq3,
 		used = NULL;
 	}
 
-	if (used && (mode != M1V2_GETSRV_RENEW)) {
-		/* Only keep the services UP, if not forced to renew */
-		struct meta1_service_url_s **up = __get_services_up(m1, used);
-		if (up && *up) {
-			*result = pack_urlv(up);
-			meta1_service_url_cleanv(up);
+	if (used) {
+		/* Check the client knows the services currently in place */
+		if (oio_str_is_set(last)) {
+			gchar *descr = meta1_url_manifest(used);
+			if (0 != strcmp(last, descr))
+				err = NEWERROR(CODE_SHARD_CHANGE,
+						"Manifest mismatch db[%s] req[%s]",
+						descr, last);
+			g_free0(descr);
+			if (err) {
+				meta1_service_url_cleanv(used);
+				return err;
+			}
+		}
+
+		/* Now there are conditions where there is no chance we need to
+		 * renew the service, so just answer them */
+		if (mode == M1V2_GETSRV_REUSE &&
+				(policy == SVCUPD_KEEP || _is_any_service_up(m1, used))) {
+			*result = pack_urlv(used);
 			meta1_service_url_cleanv(used);
 			return NULL;
 		}
-		meta1_service_url_cleanv(up);
-	}
-
-	if (used && (mode == M1V2_GETSRV_REUSE || policy == SVCUPD_KEEP)) {
-		/* Services used but unavailable, but we are told to reuse */
-		*result = pack_urlv(used);
-		meta1_service_url_cleanv(used);
-		return err;
+	} else {
+		/* No service currently in use, so we should not see accept a request
+		 * that mentions the client knowns any service */
```

Comment:
*not **see** accept* -> *not accept*
*the client know**n**s any service* -> *the client knows any service*